clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -982,8 +982,12 @@ private void throwIfProducerClosed() {
             // for other exceptions throw directly
         } catch (ApiException e) {
             log.debug("Exception occurred during message send:", e);
-            if (callback != null)
-                callback.onCompletion(null, e);
+            // producer callback will make sure to call both 'callback' and interceptor callback
+            Callback interceptCallback = new InterceptorCallback<>(callback, this.interceptors, tp);
```

Comment:
I am not sure it covers the case KafkaException is thrown in line 913.
  throw new KafkaException("Producer closed while send in progress", e);
  
  Which means tp still not initialised and has the null value.